SU2_CFD/src/solvers/CDiscAdjMeshSolver.cpp

```diff
@@ -98,18 +98,19 @@ void CDiscAdjMeshSolver::SetRecording(CGeometry* geometry, CConfig *config){
 
 void CDiscAdjMeshSolver::RegisterSolution(CGeometry *geometry, CConfig *config){
 
-  SU2_OMP_MASTER {
-    /*--- Register reference mesh coordinates ---*/
-    direct_solver->GetNodes()->Register_MeshCoord();
-  }
-  END_SU2_OMP_MASTER
-  SU2_OMP_BARRIER
+  /*--- Register reference mesh coordinates ---*/
+  direct_solver->GetNodes()->Register_MeshCoord();
+
 }
 
 void CDiscAdjMeshSolver::RegisterVariables(CGeometry *geometry, CConfig *config, bool reset){
 
+  /*--- Register boundary displacements as input.
+   * Except for FSI, where they are determined by the FEA solver. ---*/
+
+  if (config->GetFSI_Simulation()) return;
```

Comment:
Registering intermediate variables as input also causes issues with the multiple recordings.